src/setup/sound.c

```diff
@@ -142,26 +149,37 @@ static void UpdateExtraTable(TXT_UNCAST_ARG(widget),
     // Rebuild the GUS table. Start by emptying it, then only add the
     // GUS control widget if we are in GUS music mode.
 
-    TXT_ClearTable(extra_table);
-
-    if (snd_musicmode == MUSICMODE_GUS)
+    if (snd_musicmode == MUSICMODE_OPL)
     {
+        TXT_InitTable(extra_table, 2);
+        TXT_SetColumnWidths(extra_table, 19, 4);
         TXT_AddWidgets(extra_table,
-                       TXT_NewLabel("GUS patch path:"),
-                       TXT_NewFileSelector(&gus_patch_path, 30,
-                                           "Select path to GUS patches",
-                                           TXT_DIRECTORY),
-                       NULL);
-    }
+                        TXT_NewLabel("OPL type"),
+                        TXT_NewDropdownList(&opl_type, opltype_strings, 2),
+                        NULL);
+	}
+	else
```

Comment:
Formatting is screwed up here as well because you've used tab characters. Please read the HACKING file - set your editor to display tabs as 8 character indents, and to expand tabs to spaces.